spec/hydra/works/models/concerns/generic_file/characterization_spec.rb

```diff
@@ -0,0 +1,157 @@
+require 'spec_helper'
+
+describe Hydra::Works::GenericFile::Characterization do
+  # Demonstration class stand in for what downstream applications implement.
+  let(:demoClass)  do
+    class DemoClass < Hydra::Works::GenericFile::Base
+      include Hydra::Works::GenericFile::Characterization
+    end
+  end
+
+  # The is the only test context that will not mock out Characterization.characterize
+  #   Remove after development as this is more of an integration test
+  context "using fits.sh via Hydra::FileCharacterization" do
+
+    let(:filename)     { 'sample-file.pdf' }
+    let(:mimetype)     { 'application/pdf' }
+    let(:file_content) { File.open(File.join(fixture_path, filename)) }
+    let(:file)         { Hydra::PCDM::File.new.tap{ |f| f.content = file_content } }
+    let(:type)         { ::RDF::URI("http://pcdm.org/use#OriginalFile") }
+
+    subject { demoClass.new(id:'demo/01/01') }
+
+    before do
```

Comment:
> I am currently wondering about the badness level of mocking the ldp_subject.head response.

It's pretty bad, but only because #head will never -actually- return "thefavor". When you mock something it should return something with a similar API, if not a real possible return value.